filters/fadein/fadein.go

```diff
@@ -0,0 +1,252 @@
+package fadein
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	DurationName        = "fadeInDuration"
+	EaseName            = "fadeInEase"
+	EndpointCreatedName = "endpointCreated"
+)
+
+type (
+	duration time.Duration
+	ease     float64
+
+	endpointCreated struct {
+		when  time.Time
+		which string
+	}
+
+	detectedFadeIn struct {
+		when       time.Time
+		duration   time.Duration
```

Comment:
what is duration in the context of a detected fade-in ?

Reply:
It's the fade-in duration.

Reply:
starting from `when` until `duration`, so maybe `start` and `until`?

Reply:
i think this should be fine. The type name is detectedFadeIn, when stands for when was it detected, while the duration stands for whats the duration of the fade-in. 'Until' would not be good, because that rather should represent a point in time and not a duration.

Reply:
Ok!